chimedb/data_index/util.py

```diff
@@ -189,6 +192,26 @@ def parse_weatherfile_name(name):
     return name[0:8]
 
 
+def parse_miscfile_name(name):
+    """Validate and parse a misc file name.
+
+    Parameters
+    ----------
+    name : The name of the file.
+
+    Returns
+    -------
+    A tuple of containing the eight-digit serial number and the misc data
+    type.
+    """
+    m = re.match(_fmt_misc_tar, name)
+    if not m:
+        raise db.ValidationError(
+            'Bad miscellaneous file name format for "{0}".'.format(name)
+        )
+    return int(m.group(1).lstrip("0") or "0"), m.group(1)
```

Comment:
At first I was wondering why you strip the zeros off here? I guess I still don't really appreciate why it's done, but I guess we do do it for the other types.

Reply:
The immediate answer is: I copied and pasted from elsewhere in this file.

But now that you point it out, my guess is whoever wrote it the first time was thinking about C, where `strtol("013", NULL, 0)` returns 11, because the leading zero tells libc to interpret the number as octal.  Definitely not needed with Python's `int()`.

Reply:
Also that second element should be
```suggestion
    return int(m.group(1).lstrip("0") or "0"), m.group(2)
```